### PR TITLE
fix!: do not dial private addresses in browsers

### PR DESCRIPTION
**Pull request by achingbrain**
Dialling is very expensive in browsers. If we're dialling addresses that include private networks, chances are they're not going to succeed so filter them out for browsers only.

This can be re-enabled by configuring a permissive connection gater, see the upgrade guide for more information.

BREAKING CHANGE: browsers will no longer try to dial private addresses by default

### CODE REVIEW

File: src/config/connection-gater.browser.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    denyDialMultiaddr: async (multiaddr: Multiaddr) => {
+      const tuples = multiaddr.stringTuples()
+
+      if (tuples[0][0] === 4 || tuples[0][0] === 41) {
+        return Boolean(isPrivate(`${tuples[0][1]}`))
+      }
+
+      return false
+    },
+    denyInboundConnection: async () => await Promise.resolve(false),
```
Comment (wemeetagain) [inline src/config/connection-gater.browser.ts:22]: I don't htink these Promise.resolves are helpful here

Comment (achingbrain) [reply src/config/connection-gater.browser.ts:22]: Fair, this block was copied from `libp2p.ts` but I think it's been mangled by the linter auto-fix over time.

Comment (achingbrain) [reply src/config/connection-gater.browser.ts]: I've removed them

**Review by wemeetagain — APPROVED** [review]

**Comment by maschad** [comment]
Looks like there are quite a few broken tests @achingbrain

**Review by maschad — APPROVED** [review]

**Pull request merged**